### PR TITLE
Project wide GUI localization. #829

### PR DESCRIPTION
**Pull request by Bernasss12**
### The [re]start of localization
I know I could've rebased the previous branch but there were too many conflicts and since I didn't have a good way to keep track of what was done or to do I thought it would be best to start over and now I even have a [Glo](https://app.gitkraken.com/glo/board/XRTIlu_EsAAPC6PS) board, I know unfortunately it needs an account but I just logged in with GitHub.
I'm opening this draft so you can start reviewing before it is complete, if you wish to.

- `ItemDisplay.java` - remove double semicolons;
- `GuiFluidSupplierPipe.java` - utilized general `GUI` strings;
- `GuiFluidSupplierMk2Pipe.java` - utilized general `GUI` strings and repeated strings from `GuiFluidSupplierPipe.java`;
- `GuiFluidBasic.java` - added PREFIX final for consistency's sake;
- `GuiCraftingPipe` - utilized general strings;
- `GuiPipeController.java` - utilized general strings;
- `GuiProgramCompiler.java` - added prefix and completed localization;
- `GuiPowerProvider.java` - utilized general strings;
- `GuiPowerJunction.java` - utilized general strings;
- `GuiProviderPipe.java` - utilized general strings.

- `en_us.lang` - removed multiple repeated strings and re-added them as `gui.general` so multiple classes may be able to use it, also fixed some typos and helped consistency by rewriting some names in `camelCase`.

I have not been formatting (`Ctrl`+`Alt`+`L`) since I wasn't the one who wrote these files, if you want me to format them and the future ones just ask.

### CODE REVIEW

**Comment by Bernasss12** [comment]
### Wrapping up the /gui/ folder's localization, onto the sub folders.
- `GuiInvSysConnector.java` - utilized general strings;
- `GuiSatellitePipe.java` - utilized general strings and a prefix constant for consistency;
- `GuiSecurityStation.java` - utilized general strings and renamed some keys for consistency;
- `GuiStatistics.java` - utilized general strings and renamed some keys;
- `GuiSupplierPipe.java` - utilized some general strings;
- `ModuleActiveSupplier.java` - added a key and key `getter` to the `enums` so they can be translated as opposed to the way they were;

- `en_us.lang` - added more general strings and removed more duplicates.

**Comment by dblsaiko** [comment]
Please use `snake_case` in localization keys and have the mod id in them, e.g. `gui.logisticspipes.*` to prevent localization key collisions between mods.

**Comment by dblsaiko** [comment]
The PR looks good otherwise. _But again, please format your code before committing._ (if you don't want to format the entire file, select some lines and then format)

**Comment by Bernasss12** [comment]
> 
> 
> The PR looks good otherwise. _But again, please format your code before committing._ (if you don't want to format the entire file, select some lines and then format)

The only reason I haven't formatted anything is because I'm not the one coding it but sure I don't mind formatting anyways

> 
> 
> Please use `snake_case` in localization keys and have the mod id in them, e.g. `gui.logisticspipes.*` to prevent localization key collisions between mods.

About this, I just followed the system that was previously in place but I could go back and put it all according to that I guess.

**Comment by dblsaiko** [comment]
Yeah, a lot of the existing lang key names are bad, and probably there since the very early days of the mod where stuff was not as standardized yet as it is today and just haven't changed ever

**Comment by Bernasss12** [comment]
### Finished the /orderer/ folder too.
 - `GuiFluidOrderer.java` - utilized general strings and renamed some keys for consistency;
 - `GuiOrderer.java` - utilized general strings and renamed some keys for consistency;
 - `NormalMK2GuiOrderer.java` - utilized general strings and renamed some keys for consistency;
 - `NormalGuiOrderer.java` - utilized general strings and renamed some keys for consistency;
 - `GuiRequestTable.java` - utilized general strings and renamed some keys for consistency;

### Further changes
 - `LogisticsPowerProviderTileEntity` - Was this a bug? Should I revert what I changed?
 - `ModuleActiveSupplier.java` - changed the key getter;
 - Changed the syntax of all previoulsy completed `GUI` to be compliant with the current modding scene standards: `gui.logisticspipes.GUI.string_to_be_set=Key to be set`;
 - Fixed many previously made mistakes and checked every line of printed text and button to see if they were properly working _in-game_. Further testing is still welcome though 😜
 - Formatted every file I touched, finally 😂

### A few recommendations/asking for permission to:
<details><summary>Centering Strings</summary>

<p>

### I have noticed some `GUIs` still have hard coded centering for some text, I don't mind changing it to use the `drawCenteredString()` method so that translated text doesn't look misaligned.

Example: `GuiFirewall.java`

```java
mc.fontRenderer.drawString(StringUtils.translate(GuiFirewall.PREFIX + "filtered_items_are") + ":", guiLeft + 125, guiTop + 8, 0x404040);
mc.fontRenderer.drawString(StringUtils.translate(GuiFirewall.PREFIX + "providing") + ":", guiLeft + 144, guiTop + 41, 0x404040);
mc.fontRenderer.drawString(StringUtils.translate(GuiFirewall.PREFIX + "crafting") + ":", guiLeft + 146, guiTop + 74, 0x404040);
mc.fontRenderer.drawString(StringUtils.translate(GuiFirewall.PREFIX + "sorting") + ":", guiLeft + 150, guiTop + 107, 0x404040);
mc.fontRenderer.drawString(StringUtils.translate(GuiFirewall.PREFIX + "powerflow") + ":", guiLeft + 142, guiTop + 141, 0x404040);
```

image

</p>
</details>
<details><summary>Right align some text</summary>

<p>

### Some text could or _should_ be automatically right aligned instead of hard coded to accommodate  different sentence lengths in other languages. 

Example: `GuiFluidSupplierMK2Pipe.java`

```java
mc.fontRenderer.drawString(StringUtils.translate(GuiFluidSupplierMk2Pipe.PREFIX + "fluid") + ":", 25, 22, 0x404040);
mc.fontRenderer.drawString(StringUtils.translate(GuiFluidSupplierMk2Pipe.PREFIX + "partial") + ":", xSize - 176, ySize - 109, 0x404040);
```

image


</p>
</details>
<details><summary>Unused string localization</summary>

<p>

### In `GuiLogisticsSettings.java` there are commented lines, should those be removed or kept? As well as the respective translations.

```java
//useNewRendererButton = (GuiCheckBox) addButton(new GuiCheckBox(0, guiLeft + 15, guiTop + 30, 16, 16, config.isUseNewRenderer()));
//useFallbackRendererButton = (GuiCheckBox) addButton(new GuiCheckBox(0, guiLeft + 15, guiTop + 50, 16, 16, config.isUseFallbackRenderer()));	
(...)
//fontRenderer.drawString(StringUtils.translate(PREFIX + "pipe_new_renderer"), 38, 34, 0x404040);
//fontRenderer.drawString(StringUtils.translate(PREFIX + "pipe_fallback_renderer"), 38, 54, 0x404040);
```

```
gui.logisticspipes.settings.pipe_new_renderer=Use New Renderer
gui.logisticspipes.settings.pipe_fallback_renderer=Use Fallback Renderer
```

</p>

</details>
</details>
<details><summary>Unused `GuiSupplierPipe.java`</summary>

<p>

### The mentioned class doesn't seem to have any use in-game, at least that I was able to find and test ＞﹏＜

May actually my fault but tell me if that is the case.

</p>

</details>

**Comment by bziemons** [comment]
> LogisticsPowerProviderTileEntity - Was this a bug? Should I revert what I changed?

Looks like it. I will give this a quick commit. You should revert this here. Thanks for the find :)

> Formatted every file I touched, finally

finally :tada: :cake: :balloon:

**Comment by Bernasss12** [comment]
### /modules/ and /hud/ localized, /popup/ to go (for now)
As said in the commit message, both packages and their sub-packages are localized. Note that the following were not completely tested because I'm noob and don't know how to access inventories 😒 and also because some have certain translated strings that I wasn't able to find drawn on screen
 - `HUDCCBasedQuickSort.java` - can't find this one;
 - `GuiCCBasedQuickSort.java` - can't find this one either;
 - `HUDInvSysConnector.java` - can't seem to make the HUD show itself
 - `GuiFluidSupplier.java` - can't find it for the life of me
 - `GuiOreDictItemSink.java` - might have translated something I wasn't supposed to, oh well I can revert if that's the case;
 - `GuiStringBasedItemSink.java` - probably fine but I couldn't confirm myself.

**Also, I refactored a class whose name had a typo, if that is wrong I can revert that back as well 😜**

The class is question was `ActionChoisePopup.java` and is now called `ActionChoicePopup.java`, if I find any problem with this I'll fix it myself asap, so far nothing looks broken 😁

**Comment by Bernasss12** [comment]
### Hooray! Localize all the things

As of now all the classes I knew needed to be localized are finished, next order of business will be testing all the other classes that even thought I checked while I localized them every time a string got re-used and therefore renamed, a possible new bug appeared and that's what I'm aiming to filter out.

If there are any more classes that require localization please comment here.

With this the PR is almost good to go.

**Comment by Bernasss12** [comment]
### Ready for review :)
I reviewed every translated string myself but I'm only human, mistakes may come to light



<details><summary>Fixed rendering problem with HUDs</summary>
<p>

#### When shift-hovering over an OreDictItemSink HUD the crosshair would turn to a solid black square instead of a black crosshair. This was fixed on [`\gui\hud\modules\HUDOreDictItemSink.java#32`](https://github.com/RS485/LogisticsPipes/pull/1313/commits/33ccc132703dfa338e9cd36642abfbac742643b0#diff-4d0228ace9d84335cf1ffcc53dc95a6a)

Example image:


</p>
</details>

File: common/logisticspipes/gui/GuiCardManager.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package logisticspipes.gui;
 
+import net.minecraft.entity.player.EntityPlayer;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiCardManager.java]: Untouched class with formatting

File: common/logisticspipes/gui/GuiChassiPipe.java [hunk 2]
```diff
@@ -129,10 +127,10 @@ public void initGui() {
 			}
 
 			if (hasUpgradeModuleUpgarde) {
-				upgradeConfig[i*2] = addButton(new SmallGuiButton(100 + i, guiLeft + 134, guiTop + 12 + i * 20, 10, 10, "!"));
-				upgradeConfig[i*2].visible = _chassiPipe.getModuleUpgradeManager(i).hasGuiUpgrade(0);
-				upgradeConfig[i*2+1] = addButton(new SmallGuiButton(120 + i, guiLeft + 182, guiTop + 12 + i * 20, 10, 10, "!"));
-				upgradeConfig[i*2+1].visible = _chassiPipe.getModuleUpgradeManager(i).hasGuiUpgrade(1);
+				upgradeConfig[i * 2] = addButton(new SmallGuiButton(100 + i, guiLeft + 134, guiTop + 12 + i * 20, 10, 10, "!"));
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiChassiPipe.java]: Untouched class with formatting

File: common/logisticspipes/gui/GuiCraftingPipe.java [hunk 3]
```diff
@@ -1,13 +1,28 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiCraftingPipe.java]: Leave the copyright notices alone...

File: common/logisticspipes/gui/GuiChassiPipe.java [hunk 4]
```diff
@@ -1,6 +1,5 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiChassiPipe.java]: Leave the copyright alone

File: common/logisticspipes/gui/GuiFreqCardContent.java [hunk 5]
```diff
@@ -1,14 +1,14 @@
 package logisticspipes.gui;
 
+import net.minecraft.entity.player.EntityPlayer;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiFreqCardContent.java]: Untouched class with formatting

File: common/logisticspipes/gui/GuiGuideBook.java [hunk 6]
```diff
@@ -1,8 +1,8 @@
 package logisticspipes.gui;
 
-import logisticspipes.LPConstants;
-import logisticspipes.items.ItemGuideBook;
-import logisticspipes.utils.GuideBookContents;
+import java.io.IOException;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiGuideBook.java]: Untouched class with formatting

File: common/logisticspipes/gui/GuiLogisticsCraftingTable.java [hunk 7]
```diff
@@ -2,26 +2,22 @@
 
 import java.util.Arrays;
 
+import net.minecraft.client.gui.Gui;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiLogisticsCraftingTable.java]: Untouched class with formatting

File: common/logisticspipes/gui/GuiProviderPipe.java [hunk 8]
```diff
@@ -1,13 +1,21 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiProviderPipe.java]: Leave the copyright alone, dude

File: common/logisticspipes/gui/GuiSatellitePipe.java [hunk 9]
```diff
@@ -1,6 +1,5 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiSatellitePipe.java]: Again with the copyright, drop it man

File: common/logisticspipes/gui/GuiSupplierPipe.java [hunk 10]
```diff
@@ -1,6 +1,5 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/GuiSupplierPipe.java]: The copyright is not to be touched, sir

File: common/logisticspipes/gui/ItemAmountSignCreationGui.java [hunk 11]
```diff
@@ -1,15 +1,14 @@
 package logisticspipes.gui;
 
+import net.minecraft.entity.player.EntityPlayer;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/ItemAmountSignCreationGui.java]: Untouched class with formatting

File: common/logisticspipes/gui/hud/HUDProvider.java [hunk 12]
```diff
@@ -1,16 +1,16 @@
 package logisticspipes.gui.hud;
 
+import net.minecraft.client.Minecraft;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/HUDProvider.java]: Untouched class with formatting

File: common/logisticspipes/gui/hud/HUDSatellite.java [hunk 13]
```diff
@@ -1,17 +1,16 @@
 package logisticspipes.gui.hud;
 
+import net.minecraft.client.Minecraft;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/HUDSatellite.java]: Untouched class with formatting

File: common/logisticspipes/gui/hud/modules/HUDOreDictItemSink.java [hunk 14]
```diff
@@ -28,6 +29,7 @@ public void renderContent(boolean shifted) {
 		GL11.glScalef(1.0F, 1.0F, -0.00001F);
 		ItemStackRenderer.renderItemIdentifierStackListIntoGui(itemSink.getHudItemList(), null, 0, -25, -32, 3, 9, 18, 18, 100.0F, DisplayAmount.NEVER, false, shifted);
 		GL11.glScalef(1.0F, 1.0F, 1 / -0.00001F);
+		GlStateManager.enableAlpha(); // Fix cross hair rendering
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/modules/HUDOreDictItemSink.java]: This is a fix, please do it elsewhere
Other than this the class is not changed and so it does not need reformatting

File: common/logisticspipes/gui/hud/modules/HUDProviderModule.java [hunk 15]
```diff
@@ -3,19 +3,19 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import net.minecraft.client.Minecraft;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/modules/HUDProviderModule.java]: Untouched class with formatting

File: common/logisticspipes/gui/hud/modules/HUDSimpleFilterModule.java [hunk 16]
```diff
@@ -24,6 +26,7 @@ public void renderContent(boolean shifted) {
 		GL11.glScalef(1.0F, 1.0F, -0.00001F);
 		ItemStackRenderer.renderItemIdentifierStackListIntoGui(ItemIdentifierStack.getListFromInventory(module.getFilterInventory()), null, 0, -25, -32, 3, 9, 18, 18, 100.0F, DisplayAmount.NEVER, false, shifted);
 		GL11.glScalef(1.0F, 1.0F, 1 / -0.00001F);
+		GlStateManager.enableAlpha(); // Fix cross hair rendering
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/modules/HUDSimpleFilterModule.java]: Also do this in a separate pr and revert the class

File: common/logisticspipes/gui/hud/modules/HUDStringBasedItemSink.java [hunk 17]
```diff
@@ -3,15 +3,15 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import net.minecraft.client.Minecraft;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/hud/modules/HUDStringBasedItemSink.java]: Untouched class with formatting

File: common/logisticspipes/gui/modules/GuiAdvancedExtractor.java [hunk 18]
```diff
@@ -1,14 +1,16 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/GuiAdvancedExtractor.java]: Touch not the copyright notice

File: common/logisticspipes/gui/modules/GuiExtractor.java [hunk 19]
```diff
@@ -1,6 +1,5 @@
 /**
  * Copyright (c) Krapht, 2012
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/GuiExtractor.java]: Can't touch this

File: common/logisticspipes/gui/modules/GuiFluidSupplier.java [hunk 20]
```diff
@@ -1,21 +1,22 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/GuiFluidSupplier.java]: Arghh why so many times dude?

File: common/logisticspipes/gui/modules/GuiItemSink.java [hunk 21]
```diff
@@ -1,13 +1,21 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/GuiItemSink.java]: Revert this

File: common/logisticspipes/gui/modules/GuiSimpleFilter.java [hunk 22]
```diff
@@ -1,23 +1,26 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/GuiSimpleFilter.java]: Revert this

File: common/logisticspipes/gui/modules/ModuleBaseGui.java [hunk 23]
```diff
@@ -2,17 +2,17 @@
 
 import java.io.IOException;
 
+import net.minecraft.inventory.Container;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/modules/ModuleBaseGui.java]: Untouched class with formatting

File: common/logisticspipes/gui/orderer/GuiOrderer.java [hunk 24]
```diff
@@ -1,6 +1,5 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/gui/orderer/GuiOrderer.java]: Revert this

File: common/logisticspipes/utils/gui/sideconfig/SideConfigDisplay.java [hunk 25]
```diff
@@ -8,7 +8,6 @@
 import java.util.Collections;
 import java.util.List;
 
-import logisticspipes.LPBlocks;
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/sideconfig/SideConfigDisplay.java]: Untouched class with formatting

File: common/logisticspipes/utils/gui/hud/BasicHUDButton.java [hunk 26]
```diff
@@ -1,17 +1,17 @@
 package logisticspipes.utils.gui.hud;
 
-import logisticspipes.interfaces.IHUDButton;
-import logisticspipes.utils.Color;
-import logisticspipes.utils.gui.GuiGraphics;
-import logisticspipes.utils.gui.SimpleGraphics;
-
 import net.minecraft.client.Minecraft;
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/hud/BasicHUDButton.java]: Untouched class with formatting

File: common/logisticspipes/utils/gui/extention/GuiExtentionController.java [hunk 27]
```diff
@@ -5,14 +5,14 @@
 import java.util.List;
 import java.util.Map;
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/extention/GuiExtentionController.java]: Untouched class with formatting

File: common/logisticspipes/network/GuiHandler.java [hunk 28]
```diff
@@ -1,12 +1,19 @@
 package logisticspipes.network;
 
+import net.minecraft.entity.player.EntityPlayer;
```
Comment (Bernasss12) [inline common/logisticspipes/network/GuiHandler.java]: Untouched class with formatting

File: common/logisticspipes/modules/ModuleActiveSupplier.java [hunk 29]
```diff
@@ -163,18 +163,34 @@ public void setRequestFailed(boolean value) {
 	private final HashMap<ItemIdentifier, Integer> _requestedItems = new HashMap<>();
```
Comment (Bernasss12) [inline common/logisticspipes/modules/ModuleActiveSupplier.java]: Changes, might want to make a separate pr for this change

File: common/logisticspipes/gui/popup/ActionChoicePopup.java [hunk 30]
```diff
@@ -8,7 +8,7 @@
 import logisticspipes.utils.gui.GuiGraphics;
 import logisticspipes.utils.gui.SubGuiScreen;
```
Comment (Bernasss12) [inline common/logisticspipes/gui/popup/ActionChoicePopup.java]: Untouched class with formatting

File: common/logisticspipes/utils/gui/ItemDisplay.java [hunk 31]
```diff
@@ -33,6 +24,15 @@
 import org.lwjgl.input.Mouse;
 import org.lwjgl.opengl.GL11;
 
+import logisticspipes.config.Configs;
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/ItemDisplay.java]: Untouched class with formatting

File: common/logisticspipes/utils/gui/LogisticsBaseGuiScreen.java [hunk 32]
```diff
@@ -1,21 +1,20 @@
 /**
  * Copyright (c) Krapht, 2011
- *
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/LogisticsBaseGuiScreen.java]: Revert this

File: common/logisticspipes/utils/gui/LogisticsBaseTabGuiScreen.java [hunk 33]
```diff
@@ -68,7 +68,7 @@ protected void mouseClicked(int par1, int par2, int par3) throws IOException {
 		if (par3 == 0 && par1 > guiLeft && par1 < guiLeft + 220 && par2 > guiTop && par2 < guiTop + 20) {
 			par1 -= guiLeft + 3;
 			int select = Math.max(0, Math.min(par1 / 25, tabList.size() - 1));
-			if(current_Tab != select) {
+			if (current_Tab != select) {
```
Comment (Bernasss12) [inline common/logisticspipes/utils/gui/LogisticsBaseTabGuiScreen.java]: Untouched class with formatting
